simulation-service/src/main.cpp

```diff
@@ -79,6 +99,82 @@ SequenceStats calculateStats(const std::vector<double> &sequence, const std::vec
     return stats;
 }
 
+// Helper function to calculate mean of cyclic values (like hue)
+double cyclicMean(const std::vector<double>& values) {
+    double sum_sin = 0.0;
+    double sum_cos = 0.0;
+
+    for (double value : values) {
+        double angle = value * 2 * M_PI;  // Convert to radians
+        sum_sin += std::sin(angle);
+        sum_cos += std::cos(angle);
+    }
+
+    double mean_angle = std::atan2(sum_sin, sum_cos);
+    if (mean_angle < 0) mean_angle += 2 * M_PI;
+    return mean_angle / (2 * M_PI);  // Convert back to [0,1] range
+}
+
+// Helper function to calculate standard deviation of cyclic values
+double cyclicStdDev(const std::vector<double>& values, double mean) {
+    double sum_squared_diff = 0.0;
+    for (double value : values) {
+        double diff = value - mean;
+        // Handle the cyclic nature by taking the shortest distance
+        if (diff > 0.5) diff -= 1.0;
+        else if (diff < -0.5) diff += 1.0;
+        sum_squared_diff += diff * diff;
+    }
+    return std::sqrt(sum_squared_diff / values.size());
+}
+
+HSVStats calculateHSVStats(const std::vector<kivsee_render::HSV>& frames) {
+    HSVStats stats;
+    const int num_buckets = 10;
+    stats.hue_histogram.resize(num_buckets, 0);
+    stats.saturation_histogram.resize(num_buckets, 0);
+    stats.value_histogram.resize(num_buckets, 0);
+
+    std::vector<double> hues;
+    std::vector<double> saturations;
+    std::vector<double> values;
+
+    // Collect all values and build histograms
+    for (const auto& frame : frames) {
+        hues.push_back(frame.hue);
+        saturations.push_back(frame.sat);
+        values.push_back(frame.val);
+
+        // Update histograms
+        stats.hue_histogram[static_cast<int>(frame.hue * num_buckets)]++;
+        stats.saturation_histogram[static_cast<int>(frame.sat * num_buckets)]++;
+        stats.value_histogram[static_cast<int>(frame.val * num_buckets)]++;
+    }
+
+    // Calculate means
+    stats.mean_hue = cyclicMean(hues);
```

Comment:
I wonder if we should have weights based on the brightness and sat? so we know to disregard pixels that dosen't actually have color?